frn.py

```diff
@@ -50,24 +67,41 @@ def build(self, input_shape):
                                     initializer=self.beta_initializer,
                                     regularizer=self.beta_regularizer,
                                     constraint=self.beta_constraint)
+
+        if self.learnable_epsilon:
+            self.epsilon_l = self.add_weight(shape=(1,),
+                                             name='epsilon_l',
+                                             initializer=self.epsilon_l_initializer,
+                                             regularizer=self.epsilon_l_regularizer,
+                                             constraint=self.epsilon_l_constraint)
+
```

Comment:
As far as I understand - no need to set an additional epsilon variable. If learned, set `self.epsilon` initialized to the passed epsilon value.

Reply:
That's also what I would have done in the first place, but citing the paper: 

> Since eps > 0, we explored two alternative parameterizations to enforce this constraint: absolute value and exponential. While both trained well, the absolute value parameterization eps=10^-6 +|eps_l| (eps_l being a learned parameter), produced consistently better empirical results. Parameterizations of this form are also preferable because the gradient magnitudes for eps_l are independent of the value of eps.

What do you think ?